apps/src/code-studio/components/LabContainer.jsx

```diff
@@ -16,19 +16,44 @@ import moduleStyles from './LabContainer.module.scss';
 import i18n from '@cdo/locale';
 import FontAwesome from '@cdo/apps/templates/FontAwesome';
 import ErrorBoundary from './ErrorBoundary';
+import {levelsForLessonId} from '@cdo/apps/code-studio/progressReduxSelectors';
+import MusicView from '@cdo/apps/music/views/MusicView';
+import StandaloneVideo2 from '@cdo/apps/standaloneVideo2/StandaloneVideo2';
+import ProjectContainer from '@cdo/apps/labs/projects/ProjectContainer';
+import {getStandaloneProjectId} from '@cdo/apps/labs/projects/utils';
 
-const LabContainer = ({children, onError}) => {
+const LabContainer = ({onError}) => {
   const isLabLoading = useSelector(state => state.lab.isLoading);
   const isPageError = useSelector(state => state.lab.isPageError);
 
   const overlayStyle = isLabLoading
     ? moduleStyles.showingBlock
     : moduleStyles.fadeInBlock;
 
+  const currentApp = useSelector(
+    state =>
+      levelsForLessonId(state.progress, state.progress.currentLessonId).find(
+        level => level.isCurrentLevel
+      ).app
+  );
+
   return (
     <ErrorBoundary fallback={<ErrorFallbackPage />} onError={onError}>
       <div id="lab-container" className={moduleStyles.labContainer}>
-        {children}
+        <ProjectContainer channelId={getStandaloneProjectId()}>
```

Comment:
I think this file will get way too big and confusing if we have to put every lab's configuration in here. At least for now it would be good to get each lab down to 1 line (so we would extract out `MusicView` wrapped by `ProjectContainer` into some other component), and have each one look like the line for `StandaloneVideo2`

Reply:
+1. It might be nice to store some map of appName -> React class separately, so we can just look it up there and render the desired component here

Reply:
Agreed on both points.  For starters, it's lab-agnostic again.